chapters/main_components/sections/sprite_renderer/sprite_renderer.tex

```diff
@@ -0,0 +1,32 @@
+\section{Sprite Renderer}\label{sprite_renderer}
+The \texttt{SpriteRenderer} class is responsible for rendering 2D sprites.
+It takes a PNG file and a JSON file as input and creates a texture and an array of rectangles that describe the position of each sprite on the texture.
+It then uses the texture and the rectangles to render the sprites by passing the rectangle coordinates to the shader.
+It only sets the texture once to boost performance.
+
+An exemplary JSON file describing the position of the sprites on the texture is shown below:
```

Comment:
I think this has to be rewritten because it says the asme thinks as textures.tex. maybe just put a todo here